app/modules/tmr.c

```diff
@@ -132,26 +132,26 @@ static int tmr_start(lua_State* L){
 	luaL_argcheck(L, lua_isboolean(L, 2) || lua_isnil(L, 2), 2, "boolean expected");
 	int restart = lua_toboolean(L, 2);
 
-	lua_settop(L, 1);  /* ignore any args after the userdata */
 	if (tmr->self_ref == LUA_NOREF)
 		tmr->self_ref = luaL_ref(L, LUA_REGISTRYINDEX);
 
-	//we return false if the timer is not idle
+	//we return false if the timer is not idle and is not to be restarted
 	int idle = tmr->mode&TIMER_IDLE_FLAG;
 	if(!(idle || restart)){
 		lua_pushboolean(L, 0);
 	}else{
 		if (!idle) {os_timer_disarm(&tmr->os);}
 		tmr->mode &= ~TIMER_IDLE_FLAG;
 		os_timer_arm(&tmr->os, tmr->interval, tmr->mode==TIMER_MODE_AUTO);
+		lua_pushboolean(L, 1);
```

Comment:
Maybe `lua_pushboolean(L, true)` (and change the other one to `false` rather than 0?)`